### PR TITLE
Implement the XWalkViewDatabase

### PR DESCRIPTION
**Pull request by xzhan96**
This class is the equivalent of WebViewDatabase, but it only has two APIs,
hasFormData and clearFormData, they allow developers to determine whether
there is any saved data for web forms and to clear such data.

BUG=XWALK-7057

### CODE REVIEW

**Comment by xzhan96** [comment]
@rakuco I don't know why the code check always fail, could you help?

**Comment by rakuco** [comment]
Well, did you try following the advice in the log after the explanation of what was wrong?

**Comment by xzhan96** [comment]
> Running presubmit upload checks ...
Done processing /home/crwuser/build-infrastructure/slave/crosswalk-linux/build/src/xwalk/runtime/app/android/xwalk_jni_registrar.cc
Done processing /home/crwuser/build-infrastructure/slave/crosswalk-linux/build/src/xwalk/runtime/browser/android/xwalk_form_database.cc
Done processing /home/crwuser/build-infrastructure/slave/crosswalk-linux/build/src/xwalk/runtime/browser/android/xwalk_form_database.h

> ** Presubmit Warnings **
> /home/crwuser/build-infrastructure/slave/crosswalk-linux/build/src/xwalk/runtime/android/core_internal/BUILD.gn requires formatting. Please run:
>   gn format runtime/android/core_internal/BUILD.gn

I have tried "gn format runtime/android/core_internal/BUILD.gn", nothing happens

**Comment by rakuco** [comment]
> I have tried "gn format runtime/android/core_internal/BUILD.gn", nothing happens

You should _at the very least_ have seen that entire file reformatted and printed to your terminal. `gn help format` would then show that passing `--in-place` would reformat the file and replace its existing contents, after which you can see one of the entries you added to it was not alphabetically sorted.

**Comment by xzhan96** [comment]
@rakuco thank you very much for you kindly help.

**Comment by xzhan96** [comment]
@lincsoon @rakuco please help review, thanks

File: runtime/browser/android/xwalk_form_database.cc [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// Copyright 2013 The Chromium Authors. All rights reserved.
```
Comment (lincsoon) [inline runtime/browser/android/xwalk_form_database.cc]: Add Intel's copyright as well

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkFormDatabase.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// Copyright 2013 The Chromium Authors. All rights reserved.
```
Comment (lincsoon) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkFormDatabase.java]: Use Intel's copyright

**Comment by xzhan96** [comment]
@lincsoon

**Comment by xzhan96** [comment]
@lincsoon do you think it's need to merge this patch?

**Pull request merged**